### PR TITLE
Only escape arg, and only local to function.

### PR DESCRIPTION
**Pull request by whikloj**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1898

# What does this Pull Request do?

Use of `escapeshellarg` was mixed with `escapeshellcmd` in a couple functions and then the escaped argument was passed to another function where it was escaped again. 

This caused (in my case) the variable to evaluate to nothing.

# What's new?

Output escaped file path as a new variable, use that variable outside of the `escapeshellcmd` call. Don't pass the escaped file path to subsequent functions.

# How should this be tested?

In our situation this manifested when creating a JP2 derivative from a Tiff, so I would suggest trying that.

The identify process in [islandora_large_image_is_tiff](https://github.com/Islandora/islandora_solution_pack_large_image/blob/7.x/includes/derivatives.inc#L495) is called from [islandora_large_image_is_uncompressed_tiff](https://github.com/Islandora/islandora_solution_pack_large_image/blob/7.x/includes/derivatives.inc#L478) with a value already escaped.

When it is escaped a second time, our value became `''` which meant that identify was not given an argument. It didn't seem to exit, but rather wait.

This PR isolates the escaped file path to each function and ensures that the variable escaped by `escapeshellarg` is not re-escaped by `escapeshellcmd`.

Example:
* Does this change require documentation to be updated? no
* Does this change add any new dependencies? np
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? I hope so

# Interested parties
@nigelgbanks @qadan - as they have more experience with escapeshellarg, in case I have missed something.

### CODE REVIEW

**Comment by DiegoPino** [comment]
@whikloj any idea on when this was introduced? I guess this is a relatively new bug,  it would be good to know who/versions may be affected, because if reproducible and not and edge case, it is basically blocking all derivatives in this SP from TIFFs to happen.

**Comment by nigelgbanks** [comment]
Figured I was responsible as I was mucking about with this section some months ago, but this is what I got from the git blame.

https://github.com/Islandora/islandora_solution_pack_large_image/pull/160

So it was in the last 2 months.

This pull is inline the other recent fixes that have been tested:

https://github.com/Islandora/islandora_solution_pack_large_image/pull/163

That is the use of `escapeshellarg` and appending it to `escapeshellcmd` rather than embedding it within it.

Might be time for a small audit of calls to the shell?

**Comment by DiegoPino** [comment]
@nigelgbanks its ok, there are many many chained to calls everywhere, so kinda normal these things happen 😄 . So wondering, if that pull introduced the error, does this pull, which should fix it, would still solve your original issue with whitespaces? Or do we need to go for some intermediate solution? Thanks!

**Comment by nigelgbanks** [comment]
Have to see the input file to know for sure, but this issue is the same as @qadan  resolved in https://github.com/Islandora/islandora_solution_pack_large_image/pull/163, just different functions. I think this pull is safe, I don't have time to test at the moment though, I'll do some testing tonight though.

**Comment by qadan** [comment]
Haha goodness yeah, we seem to be doing this a lot lately 😅 but yes, I think in every situation it's been most safe to do it like this, i.e., concatenating an `escapeshellcmd()`'d command with an `escapeshellarg()`'d argument.

Took a quick peek - we appear to do this in islandora_scholar's [bibutils](https://github.com/Islandora/islandora_scholar/blob/634ba0ac9078f27c252ba2ce71f899e396e3bbb3/modules/bibutils/includes/utilities.inc#L24-L32) submodule in a completely benign way, [islandora_ocr](https://github.com/Islandora/islandora_ocr/blob/fa9de96d32600db4be6055c7a22b7b16f9e923ac/includes/derivatives.inc#L160-L162) in a fairly benign way (actually two instance in there), [islandora_paged_content](https://github.com/Islandora/islandora_paged_content/blob/032bc65eccf2fce1cbe8bc849fd080ea685483d7/includes/utilities.inc#L1142) in a way we might want to address (other instances there seem more benign), and [islandora_solution_pack_web_archive](https://github.com/Islandora/islandora_solution_pack_web_archive/blob/8731695ecbeae2d5433ea3dc562f46b9c826fb3b/includes/derivatives.inc#L398) (and [here](https://github.com/Islandora/islandora_solution_pack_web_archive/blob/8731695ecbeae2d5433ea3dc562f46b9c826fb3b/includes/derivatives.inc#L415)) in a way that may be a similar ticking timebomb to the ones we've been addressing recently.

So I'd be in agreement that it's perhaps time for an "audit-and-repair-escapeshellcmd/escapeshellarg calls" ticket, given that there's at least three functions across two modules that could get tripped up by special filenames

**Review by DiegoPino — APPROVED** [review]
Approving this pull. @Islandora/7-x-1-x-committers please comment if there are concerns. Will merge before the end of the day if not. Thanks

**Comment by nigelgbanks** [comment]
Sorry I took so long to test! But it looks like there is some problem there was a regressions in identify files.


```php
module_load_include('inc', 'islandora_large_image', 'includes/derivatives');
$files = array(
 '/tmp/sample.tif',
 '/tmp/sample space.tif',
 '/tmp/sample:!@#$%^&*()-=;.tif',
 '/tmp/sample.jp2',
 '/tmp/sample space.jp2',
 '/tmp/sample:!@#$%^&*()-=;.jp2',
);
foreach ($files as $file) {
  dsm(islandora_large_image_is_uncompressed_tiff($file) ? 'TRUE' : 'FALSE', "Uncompressed TIFF: $file");
  dsm(islandora_large_image_is_tiff($file) ? 'TRUE' : 'FALSE', "TIFF: $file");
  dsm(islandora_large_image_is_jp2($file) ? 'TRUE' : 'FALSE', "JP2: $file");
}
```

**7.x => Yields:**
```
# Correct
Uncompressed TIFF: /tmp/sample.tif => FALSE
TIFF: /tmp/sample.tif => TRUE
JP2: /tmp/sample.tif => FALSE

# Correct
Uncompressed TIFF: /tmp/sample space.tif => FALSE
TIFF: /tmp/sample space.tif => TRUE
JP2: /tmp/sample space.tif => FALSE

# Not correct
Uncompressed TIFF: /tmp/sample:!@#$%^&*()-=;.tif => FALSE
TIFF: /tmp/sample:!@#$%^&*()-=;.tif => FALSE
JP2: /tmp/sample:!@#$%^&*()-=;.tif => FALSE

# Correct
Uncompressed TIFF: /tmp/sample.jp2 => FALSE
TIFF: /tmp/sample.jp2 => FALSE
JP2: /tmp/sample.jp2 => TRUE

# Not correct
Uncompressed TIFF: /tmp/sample space.jp2 => FALSE
TIFF: /tmp/sample space.jp2 => FALSE
JP2: /tmp/sample space.jp2 => FALSE

# Correct
Uncompressed TIFF: /tmp/sample:!@#$%^&*()-=;.jp2 => FALSE
TIFF: /tmp/sample:!@#$%^&*()-=;.jp2 => FALSE
JP2: /tmp/sample:!@#$%^&*()-=;.jp2 => TRUE
```

**Pull  => Yields:**
```
# Not correct
Uncompressed TIFF: /tmp/sample.tif => FALSE
TIFF: /tmp/sample.tif => FALSE
JP2: /tmp/sample.tif => FALSE

# Not correct
Uncompressed TIFF: /tmp/sample space.tif => FALSE
TIFF: /tmp/sample space.tif => FALSE
JP2: /tmp/sample space.tif => FALSE

# Not correct
Uncompressed TIFF: /tmp/sample:!@#$%^&*()-=;.tif => FALSE
TIFF: /tmp/sample:!@#$%^&*()-=;.tif => FALSE
JP2: /tmp/sample:!@#$%^&*()-=;.tif => FALSE

# Not correct
Uncompressed TIFF: /tmp/sample.jp2 => FALSE
TIFF: /tmp/sample.jp2 => FALSE
JP2: /tmp/sample.jp2 => FALSE

# Not correct
Uncompressed TIFF: /tmp/sample space.jp2 => FALSE
TIFF: /tmp/sample space.jp2 => FALSE
JP2: /tmp/sample space.jp2 => FALSE

# Not correct
Uncompressed TIFF: /tmp/sample:!@#$%^&*()-=;.jp2 => FALSE
TIFF: /tmp/sample:!@#$%^&*()-=;.jp2 => FALSE
JP2: /tmp/sample:!@#$%^&*()-=;.jp2 => FALSE
```

File: includes/derivatives.inc
Comment (nigelgbanks) [inline includes/derivatives.inc]: Ah needs a space between the format and last argument like so:

```php
$compression = exec(escapeshellcmd("$identify -format \"%C\" ") . $escaped_file);
```

File: includes/derivatives.inc
Comment (nigelgbanks) [inline includes/derivatives.inc]: Needs a space between the format and last argument like so:

```php
$codec = exec(escapeshellcmd("$identify -format \"%m\" ") . $escaped_file);
```

File: includes/derivatives.inc
Comment (nigelgbanks) [inline includes/derivatives.inc]: Needs a space between the format and last argument like so:

```php
$codec = exec(escapeshellcmd("$identify -format \"%m\" ") . $escaped_file);
```

**Review by nigelgbanks — CHANGES_REQUESTED** [review]

**Comment by whikloj** [comment]
Thanks @nigelgbanks, I have made the requested changes

**Comment by DiegoPino** [comment]
@whikloj @nigelgbanks thanks for checking this. I will run the same tests @nigelgbanks executed  to make sure nothing fails.

**Comment by nigelgbanks** [comment]
@DiegoPino since I already had the enviroment up this morning, I can give a quick test.

```
# Correct.
Uncompressed TIFF: /tmp/sample.tif => FALSE
TIFF: /tmp/sample.tif => TRUE
JP2: /tmp/sample.tif => FALSE

# Correct.
Uncompressed TIFF: /tmp/sample space.tif => FALSE
TIFF: /tmp/sample space.tif => TRUE
JP2: /tmp/sample space.tif => FALSE

# Correct.
Uncompressed TIFF: /tmp/sample:!@#$%^&*()-=;.tif => FALSE
TIFF: /tmp/sample:!@#$%^&*()-=;.tif => TRUE
JP2: /tmp/sample:!@#$%^&*()-=;.tif => FALSE

# Correct.
Uncompressed TIFF: /tmp/sample.jp2 => FALSE
TIFF: /tmp/sample.jp2 => FALSE
JP2: /tmp/sample.jp2 => TRUE

# Correct.
Uncompressed TIFF: /tmp/sample space.jp2 => FALSE
TIFF: /tmp/sample space.jp2 => FALSE
JP2: /tmp/sample space.jp2 => TRUE

# Correct.
Uncompressed TIFF: /tmp/sample:!@#$%^&*()-=;.jp2 => FALSE
TIFF: /tmp/sample:!@#$%^&*()-=;.jp2 => FALSE
JP2: /tmp/sample:!@#$%^&*()-=;.jp2 => TRUE
```

Everything looks good to go.

**Review by nigelgbanks — APPROVED** [review]

**Comment by nigelgbanks** [comment]
I'll leave it to you @DiegoPino to merge, if you're happy with everything.

**Comment by DiegoPino** [comment]
@nigelgbanks @whikloj only happy if you both are happy! Looks good and _really_ thanks a lot for re-testing.

**Pull request merged**
